pkg/secrets/service/manager.go

```diff
@@ -53,6 +70,30 @@ func (c *serviceCrypter) EncryptValue(ctx context.Context, plaintext string) (st
 	return base64.StdEncoding.EncodeToString(ciphertext), nil
 }
 
+func (c *serviceCrypter) SupportsBulkEncryption(ctx context.Context) bool {
+	supports, _ := c.supportsBulkEncryption.Result(ctx)
+	return supports
+}
+
+func (c *serviceCrypter) BulkEncrypt(ctx context.Context, plaintexts []string) ([]string, error) {
+	if !c.SupportsBulkEncryption(ctx) {
+		return nil, errors.New("bulk encryption is not supported")
```

Comment:
Should we fall back to regular encryption instead of failing here in this case?  Could self-hosted service instances potentially not support bulk encryption yet, but the CLI does?  Just returning an error seems a bit unfriendly in that case.

Reply:
The contract is you shouldn't call BulkEncrypt if SupportsBulkEncrypt was false. I think this is fine.

Reply:
The original design was to avoid the 2 pass if we can't batch .. but now we're doing the transaction style we could just always use the bulk method and implement it on all crypters using a loop if nothing else ... so then we could drop the "supports" method perhaps 🤔

Reply:
If that's just as performant that sounds simpler

Reply:
I think this is worth doing, because (I think) if you do that, all calls in that piece of code go through `EnqueueEncryption`, which takes a pointer to a secret. That means that you can (maybe?) get rid of the whole "if this is a caching crypter" stuff, because caching crypters can just wrap `EnqueueEncryption` to populate the cache using the secret pointer as a key. Maybe...

Reply:
Not quite I don't think ... but we can always use the bulk approach even when not using the service and then encrypt with a degree of parallelism for the third party and local crypters.

Separately, I've also started looking at re-working the decryption side to match this pattern too, which I belive will lead to plenty of scenarios where we can completely skip calling the encrypter if no secrets have been changed:
- https://github.com/pulumi/pulumi/pull/18743